### ISSUE TITLE
Is it suitable to add "--targetRNVersion" capability to CodePush

### ISSUE THREAD

**Issue by NicolasSchwarzer** [Microsoft/code-push#607]
We're trying to use CodePush to manage our react native bundles' publish, and our app will upgrade react native version as the framework upgrades, there maybe a risk that if the app's react native version goes faster than the bundle's upgrading and publish.

For example, now the app and bundle are both of react native 0.57, once the app upgrades to 0.58 in the future, and if we forgot to upgrade js bundle, it may cause a crash on user side. In such condition, we think CodePush server should compare react native versions, and tell the app there's no package available, so the app may just load some default pages locally.

So I think if there's a "--targetRNVersion" capability, may solve this problem.

**Comment by alexandergoncharov** [comment]
Hi @NicolasSchwarzer ,
Thanks for reporting!

It is interesting questions. 
At first just small clarification:
CodePush can change only js part of your app and assets. CodePush **can not** change any binary files.
CodePush has ` -t|--target-binary-version` option for providing binary version of your app. If users's app has other binary version than you provided in this option then this update will be ignored.

In your case when you update your RN version you should make new binary apk/ipa file and provide it to your users as new version of RN can contains some changes in native code part(like Java or Objective-c).
So, the main idea is if CodePush updates contains releases with different RN version then it should be target to different binary version.

Could you please let me know if it was helpful and make sense for you or you have any questions.

Thanks,
Alexander

**Comment by NicolasSchwarzer** [comment]
Hi @alexandergoncharov ,
Thanks for your answer!

This issue is just my suggestion, indeed we can avoid the example problem I proposed by using `-t|--target-binary-version` very cautiously, and this way could also work for us. If we had encountered the same problem and there was no solution in the future, I would reopen this issue and on that condition, we could make this requirement more clear and demanded.

I will close this issue.

Thanks,
Nicolas
